### PR TITLE
Better document remove_small_objects behaviour: int vs bool

### PR DESCRIPTION
**Pull request by Phlya**
Fixing documentation according to https://github.com/scikit-image/scikit-image/issues/2825

Also added an example showing the different in behaviour together with the warning that I get.

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by stefanv** [comment]
How about a slightly different approach: we change the docstring to make it explicit that only labeled arrays should ever be provided.  We then keep the special behavior for boolean arrays, but warn that we are labeling the array before use.

This should simplify things quite a bit.

File: skimage/morphology/misc.py [hunk 1]
```diff
@@ -45,20 +45,24 @@ def _check_dtype_supported(ar):
                         "Got %s." % ar.dtype)
 
 def remove_small_objects(ar, min_size=64, connectivity=1, in_place=False):
-    """Remove connected components smaller than the specified size.
+    """Remove objects smaller than the specified size.
+    Expects ar to be an array with labelled objects, and removes objects
```
Comment (stefanv) [inline skimage/morphology/misc.py]: Insert newline after first descriptive line (NumpyDoc format)

File: skimage/morphology/misc.py [hunk 2]
```diff
@@ -45,20 +45,24 @@ def _check_dtype_supported(ar):
...
+    smaller than min_size. If ar is bool, the image is first labelled.
```
Comment (stefanv) [inline skimage/morphology/misc.py]: labeled

Comment (stefanv) [reply skimage/morphology/misc.py]: ar -> ``` `ar` ```

File: skimage/morphology/misc.py [hunk 3]
```diff
@@ -45,20 +45,24 @@ def _check_dtype_supported(ar):
...
-        The ints must be non-negative.
+        The array containing the objects of interest. If the array type is
+        int, the ints must be non-negative.
     min_size : int, optional (default: 64)
-        The smallest allowable connected component size.
+        The smallest allowable object size.
     connectivity : int, {1, 2, ..., ar.ndim}, optional (default: 1)
-        The connectivity defining the neighborhood of a pixel.
+        The connectivity defining the neighborhood of a pixel. Used during
+        labelling if ar is bool.
```
Comment (stefanv) [inline skimage/morphology/misc.py]: ar -> ``` `ar` ```

File: skimage/morphology/misc.py [hunk 4]
```diff
@@ -45,20 +45,24 @@ def _check_dtype_supported(ar):
...
     in_place : bool, optional (default: False)
-        If `True`, remove the connected components in the input array itself.
+        If `True`, remove the objects in the input array itself.
```
Comment (stefanv) [inline skimage/morphology/misc.py]: If True or ``` If ``True`` ```

File: skimage/morphology/misc.py [hunk 5]
```diff
@@ -92,6 +96,15 @@ def remove_small_objects(ar, min_size=64, connectivity=1, in_place=False):
     >>> d = morphology.remove_small_objects(a, 6, in_place=True)
     >>> d is a
     True
+    >>> e = morphology.remove_small_objects(a.astype(int), 6)
+    /usr/local/lib/python3.5/dist-packages/skimage/morphology/misc.py:135:
```
Comment (stefanv) [inline skimage/morphology/misc.py]: This won't pass the test suite.  Let's remove this example.

Comment (Phlya) [reply skimage/morphology/misc.py]: OK, that's a shame, I would have found this very useful when trying to figure out my problem...

**Comment by stefanv** [comment]
I think I meant that we should warn in the code, not in the documentation.  The docs should only describe one way of input: a labeled array.

**Comment by jni** [comment]
@stefanv I kind of like the bool polymorphism, but I see your point. However if we really want to discourage the use of bool then I suggest we simply deprecate it. @Phlya as one of the people bitten by the dual behaviour, what do you prefer?

**Comment by stefanv** [comment]
@jni Agreed.  For consistency, let's rename the parameter `label_image` and raise when it is of dtype `bool`.

**Pull request merged**